### PR TITLE
enh(article): add dbee

### PR DESCRIPTION
**Pull request by asdf8601**

### CODE REVIEW

File: content/articles/2021-08-01-dbee-nvim.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          """
+    out = pd.read_sql(qry, conn)
+    return out
+```
+
+We can setup our connection by going to `cn_str` line, selecting the content
+inside the quotes and executing `:'<,'>DBeeSetConnection`. This will print out
+the following message: `Configurated
+sqlite:////home/mgreco/gitlab/mmngreco/dbee.nvim/tests/chinook.db`, after that
+we are ready to make queries to the database.
```
Comment (asdf8601) [inline content/articles/2021-08-01-dbee-nvim.md]: ``
```suggestion
we are ready to make queries to that database.
```

File: content/articles/2021-08-01-dbee-nvim.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    conn = engine.connect()
+    qry = """
+          SELECT title
+          FROM albums;
+          """
+    out = pd.read_sql(qry, conn)
+    return out
+```
+
+We can setup our connection by going to `cn_str` line, selecting the content
```
Comment (asdf8601) [inline content/articles/2021-08-01-dbee-nvim.md]:
```suggestion
We can setup our connection by going to the line where `cn_str` is and select the content
```

**Review by asdf8601 — COMMENTED** [review]
Some improvements

File: content/articles/2021-08-01-dbee-nvim.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+# My first vim plugin!
+
+Have you ever had the necessity of check a query while you are writing a python
```
Comment (emvalbuena) [inline content/articles/2021-08-01-dbee-nvim.md]: Have you ever had the necessity to check a query while you are writing a python

File: content/articles/2021-08-01-dbee-nvim.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+wrapper? If so, this post is for you.
+
+Many times I had the necessity of checking a query while I'm writing a function,
+debugging or testing it. That's why I wrote my first vim (well, actually a
```
Comment (emvalbuena) [inline content/articles/2021-08-01-dbee-nvim.md]: Many times I have had the necessity of checking a query while I'm writing,
debugging or testing a function.

File: content/articles/2021-08-01-dbee-nvim.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+nvim) plugin called [`dbee.nvim`](https://github.com/mmngreco/dbee.nvim).
+
+## My using case
```
Comment (emvalbuena) [inline content/articles/2021-08-01-dbee-nvim.md]: ## My use case

File: content/articles/2021-08-01-dbee-nvim.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+I have a virtual environment with `SQLAlchemy` and others Python dependencies
+and then I want to check a query from my project. So, I have to open `DBeaver`
+or another SQL client copy-paste the query there and see the result. This
+process is very annoying.
```
Comment (emvalbuena) [inline content/articles/2021-08-01-dbee-nvim.md]: I have a virtual environment with `SQLAlchemy` and others Python dependencies,
and then I want to check a query from my project. So, I have to open `DBeaver`,
or another SQL client, copy-paste the query there and see the result. This 
process can be very annoying.

File: content/articles/2021-08-01-dbee-nvim.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+Why I have to open an external program when I already had all what I need in my
```
Comment (emvalbuena) [inline content/articles/2021-08-01-dbee-nvim.md]: Why do I have to open an external program when I already have all that I need in my

File: content/articles/2021-08-01-dbee-nvim.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+virtual environment?
+
+The idea is define a `engine` using `SQLAlchemy` using a connection string, and
+then through `pandas` fetch the data using that engine. Pretty simple, isn't
+it?
```
Comment (emvalbuena) [inline content/articles/2021-08-01-dbee-nvim.md]: The idea is to define an `engine` using `SQLAlchemy` with a connection string, and
then through `pandas` fetch the data using that engine. Pretty simple, isn't
it?

File: content/articles/2021-08-01-dbee-nvim.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+I'm using the following maps:
+
+```
+vnoremap <C-q> :DBeeQuery<cr>
+vnoremap <C-c><C-c> :DBeeSetConnection<cr><cr>
+```
+
+## Example
+
+Suppose we are writing the following python function:
```
Comment (emvalbuena) [inline content/articles/2021-08-01-dbee-nvim.md]: Let's suppose that we are writing the following python function:

File: content/articles/2021-08-01-dbee-nvim.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+We can setup our connection by going to `cn_str` line, selecting the content
+inside the quotes and executing `:'<,'>DBeeSetConnection`. This will print out
+the following message: `Configurated
+sqlite:////home/mgreco/gitlab/mmngreco/dbee.nvim/tests/chinook.db`, after that
+we are ready to make queries to the database.
+
+Now, we are ready to select the query inside the quotes and call to
+`:'<,'>DBeeQuery`. This should open a new temporary buffer with the connection
+string, the query used and the output on it. You can easily close the buffer by
```
Comment (emvalbuena) [inline content/articles/2021-08-01-dbee-nvim.md]: string, the selected query and the output fetched from it. You can easily close the buffer by

**Review by emvalbuena — CHANGES_REQUESTED** [review]
Hi Max, 

As requested I have had a look ;)
Pretty useful, I should one day follow you in your path with VIM.

Cheers

File: content/articles/2021-08-01-dbee-nvim.md (hunk 9 above)
Comment (asdf8601) [reply content/articles/2021-08-01-dbee-nvim.md]: This is a good one, I realized that I have problems guessing when I should omit "that".

**Review by asdf8601 — COMMENTED** [review]
@KikeM Many thanks for your feedback. In case you don't know, you can use the "suggestions" button which is perfect for this purpose.

image

Thanks! ;-)

**Pull request merged**